int8 X
int8 Y
float32 theta
